sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/TimerStateImpl.scala

```diff
@@ -188,9 +187,12 @@ class TimerStateImpl(
 
   /**
    * Function to get all the registered timers for all grouping keys
+   * @param expiryTimestampMs Threshold for expired timestamp in milliseconds, this function
```

Comment:
Nit: maybe add a small comment here mentioning that we perform a range scan and stop iterating once the key row timestamp exceeds the threshold

Reply:
This class is not user-facing, right? If it is, I'd suggest avoiding implementation detail. Looks like as it doesn't seem to be an user facing, but just to double check.

Reply:
Thanks for pointing this out (I did not realize this before but looks like we did the right thing)! We've also left out those implementation details in the user facing function `getExpiredTimers()` in `StatefulProcessorHandleimpl` here:
https://github.com/jingz-db/spark/blob/36ab93519ae1ab43e20926ae47406cff668a3b10/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StatefulProcessorHandleImpl.scala#L166-L174